### PR TITLE
feat: Support installing Thunderstore packages

### PR DESCRIPTION
**Pull request by GeckoEidechse**
- [x] Best to have #417 first cause otherwise installed mods cannot be read.
- [x] Safety check that mod is actually correctly formatted
- [x] Uninstall older version of package if installed
- [x] ~~Wait for libthermite stabilisation~~ Current version seems stable enough

Part of #415

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
For testing I used
- `Alystrasz-Parkour-0.1.1` as a mod that should pass instal
- `ScureX-ModCreator-1.0.0` as a mod that should fail install

**Comment by GeckoEidechse** [comment]
Also tested updating an existing mod where a newer version is broken. It correctly didn't install new version and it kept old version.

**Comment by GeckoEidechse** [comment]
CI error is due to clippy due to libthermite marking certain functions as deprecated early. Will look into how to best address but shouldn't prevent actually testing ^^"

File: src-tauri/src/mod_management/legacy.rs [hunk 1]
```diff
@@ -115,6 +115,42 @@ pub fn parse_installed_mods(
     Ok(mods)
 }
 
+/// Deletes all legacy packages that match in author and mod name
```
Comment (Alystrasz) [inline src-tauri/src/mod_management/legacy.rs:118]: Could you be more precise regarding to what is a legacy package?

File: src-tauri/src/mod_management/mod.rs [hunk 2]
```diff
@@ -407,6 +408,93 @@ async fn get_mod_dependencies(thunderstore_mod_string: &str) -> Result<Vec<Strin
...
+        }
+    }
+
+    for directory in directories {
+        let folder_name = directory.file_name().unwrap().to_str().unwrap();
+        let ts_mod_string_from_folder: ParsedThunderstoreModString = match folder_name.parse() {
+            Ok(res) => res,
+            Err(err) => {
+                log::warn!("{err}");
+                continue;
```
Comment (Alystrasz) [inline src-tauri/src/mod_management/mod.rs:448]: Why continuing if there was an error?

File: src-tauri/src/mod_management/legacy.rs (hunk 1 above)
Comment (GeckoEidechse) [reply src-tauri/src/mod_management/legacy.rs:118]: Sure. "Legacy package" is what I call a Thunderstore package installed into `mods` the old-fashioned way, i.e. how we did it up to now as opposed to using the `packages` folder.

Should I update the comment with something like that? ^^

Comment (Alystrasz) [reply src-tauri/src/mod_management/legacy.rs:118]: Yep, please add this to the documentation :)

File: src-tauri/src/mod_management/mod.rs (hunk 2 above)
Comment (Alystrasz) [reply src-tauri/src/mod_management/mod.rs:448]: Please write this in the documentation as well :)

**Review by Alystrasz — APPROVED** [review]
Omitting missing documentation, the feature works as expected (both in installing correcly-formatted mods and preventing install badly-formatted ones) on Windows 10 and Ubuntu 23.04.

File: src-tauri/src/mod_management/legacy.rs (hunk 1 above)
Comment (GeckoEidechse) [reply src-tauri/src/mod_management/legacy.rs:118]: Done with 448c6f343984b48b3ba7e1b9faaab54651b758f9

File: src-tauri/src/mod_management/mod.rs (hunk 2 above)
Comment (GeckoEidechse) [reply src-tauri/src/mod_management/mod.rs:452]: Done with e4f995b1f8e25ed6cddae2672ea3e79da80e55ee

**Pull request merged**